inst/templates/utils_template.R

```diff
@@ -163,36 +163,45 @@
 #' \code{pkgs}, and throws an exception with an informative error message if
 #' that is not the case.
 #' 
-#' @param pkgs Character vector with package names.
-#' @param bioc Logical scalar. If \code{TRUE}, suggest to install the missing
-#'   package(s) using \code{BiocManager} instead of \code{install.packages}.
+#' @param pkgs Character vector with package names. Can be either just a
+#'   package name or a string of the form \code{"githubuser/packagename"} for
+#'   packages hosted on GitHub.
+#' @param suggestInstallation Logical scalar. If \code{TRUE}, include an
+#'   expression to install the missing package(s) as part of the generated
+#'   error message.
 #' 
 #' @author Michael Stadler, Charlotte Soneson
 #' 
 #' @noRd
 #' @keywords internal
-.assertPackagesAvailable <- function(pkgs, bioc = TRUE) {
+.assertPackagesAvailable <- function(pkgs, suggestInstallation = TRUE) {
     stopifnot(exprs = {
         is.character(pkgs)
-        is.logical(bioc)
-        length(bioc) == 1L
+        is.logical(suggestInstallation)
+        length(suggestInstallation) == 1L
     })
     
-    avail <- unlist(lapply(pkgs,
+    avail <- unlist(lapply(sub("^[^/]+/", "", pkgs),
                            function(pkg) {
                                requireNamespace(pkg, quietly = TRUE)
                            }))
     
     if (any(!avail)) {
         caller <- deparse(sys.calls()[[sys.nframe() - 1]])
         callerfunc <- sub("\\(.+$", "", caller)
-        stop("The package", ifelse(sum(!avail) > 1, "s '", " '"),
-             paste(pkgs[!avail], collapse = "', '"), "' ",
-             ifelse(sum(!avail) > 1, "are", "is"), " required for ",
-             callerfunc, "(), but not installed.\n",
-             "Install ", ifelse(sum(!avail) > 1, "them", "it"), " using:\n",
-             ifelse(bioc, "BiocManager::install(c(\"", "install.packages(c(\""),
-             paste(pkgs[!avail], collapse = "\", \""), "\"))", call. = FALSE)
+        haveBioc <- requireNamespace("BiocManager", quietly = TRUE)
+        msg <- paste0("The package", ifelse(sum(!avail) > 1, "s '", " '"),
+                      paste(pkgs[!avail], collapse = "', '"), "' ",
```

Comment:
Would it make sense to just print out the package name here? I.e. only `pkgname` rather than `username/pkgname` for GitHub packages?

Reply:
Yes, you are right, `username/pkgname` is only needed in the installation expression.